### PR TITLE
Enable calltarget by default on .NET 4.6+ runtimes

### PR DESCRIPTION
**Pull request by kevingosse**

### CODE REVIEW

File: src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp [hunk 1]
```diff
@@ -96,6 +96,15 @@ CorProfiler::Initialize(IUnknown* cor_profiler_info_unknown) {
     return E_FAIL;
   }
 
+  // get ICorProfilerInfo6 for net452+
+  ICorProfilerInfo6* info6;
```
Comment (lucaspimentel) [inline src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp:100]: This comment seems wrong. Is it net46+?

Comment (kevingosse) [reply src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp:100]: Oh, right

Comment (kevingosse) [reply src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp:100]: Fixed in https://github.com/DataDog/dd-trace-dotnet/pull/1542/commits/338ec8db17cd8aa5198215743e2ce066ead20672

File: src/Datadog.Trace.ClrProfiler.Native/environment_variables_util.h [hunk 2]
```diff
@@ -46,14 +46,14 @@ bool EnableInlining(bool defaultValue) {
                        defaultValue);
 }
 
-bool IsCallTargetEnabled() {
+bool IsCallTargetEnabled(bool is_net46_or_greater) {
```
[the remark this answers is not in the document]
Comment (kevingosse) [reply src/Datadog.Trace.ClrProfiler.Native/environment_variables_util.h]: Fixed in https://github.com/DataDog/dd-trace-dotnet/pull/1542/commits/338ec8db17cd8aa5198215743e2ce066ead20672

**Review by lucaspimentel — APPROVED** [review]

**Review by tonyredondo — APPROVED** [review]

**Review by zacharycmontoya — APPROVED** [review]

**Pull request merged**